### ISSUE TITLE
support memcache flags during snapshotting and replication

### ISSUE THREAD

**Issue by romange** [dragonflydb/dragonfly#1758]
memcache has additional `flags` property for each its entries. 
We maintain it in a separate datastructure `mc_flags`.  

1. SAVE/LOAD procedures do not support them because it's not part of the standard Redis datastrcutures
2. Also the replication ignores them. 

We have a user that runs Dragonfly in memcached mode and H/A and asks for those features to improve the availability of their deployment.

**Comment by royjacobson** [comment]
Is the journaling subsystem supporting memcache commands? We might need to fix this as well.

**Comment by Abhra303** [comment]
What is the latest update on the issue?

**Comment by sbkoto** [comment]
not fixed on 1.19.0

```
set item 4 3600 4
test
STORED
get item
VALUE item 4 4
test
END

```

after reload dragonfly and load from snapshot:
